pkgs/development/libraries/science/math/nccl/tests.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, which
+, cudaPackages
+, mpiSupport ? false
+, mpi
+}:
+
+assert mpiSupport -> mpi != null;
+
+with cudaPackages;
+
+cudaPackages.backendStdenv.mkDerivation rec {
+
+  pname = "nccl-tests";
+  version = "2.13.6";
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-3gSBQ0g6mnQ/MFXGflE+BqqrIUoiBgp8+fWRQOvLVkw=";
+  };
+
+  nativeBuildInputs = [ which cuda_nvcc ];
+
+  buildInputs = [
+    cuda_cudart
+    nccl
+  ] ++ lib.optional mpiSupport mpi;
+
+  makeFlags = [
+    "CUDA_HOME=${cudatoolkit}"
```

Comment:
```suggestion
    "CUDA_HOME=${cudaPackages.cuda_nvcc}"
```

Move away from `with cudaPackages`.

Additionally, we don't need to bring `cudatoolkit` into the closure -- all this project really needs is `nvcc`.